src/AutoRest.CSharp/Common/Input/CadlInputConstantConverter.cs

```diff
@@ -0,0 +1,118 @@
+        {
+            var isFirstProperty = id == null;
+            Object? value = null;
+            InputType? type = null;
+            while (reader.TokenType != JsonTokenType.EndObject)
+            {
+                var isKnownProperty = reader.TryReadReferenceId(ref isFirstProperty, ref id)
+                    || reader.TryReadWithConverter(nameof(InputConstant.Type), options, ref type)
+                    || TryReadConstantValue(ref reader, nameof(InputConstant.Value), options, type, ref value);
+
+                if (!isKnownProperty)
+                {
+                    reader.SkipProperty();
+                }
+            }
+
+            type = type ?? throw new JsonException("InputConstant must have type");
+
+            value = value ?? throw new JsonException("InputConstant must have value");
+
+            var constant = new InputConstant(value, type);
+            if (id != null)
+            {
+                resolver.AddReference(id, constant);
+            }
+            return constant;
+        }
+
+        public static bool TryReadConstantValue(ref Utf8JsonReader reader, string propertyName, JsonSerializerOptions options, InputType? type, ref Object? value)
+        {
+            if (type == null)
+            {
+                throw new JsonException("Must place type ahead of value.");
+            }
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException();
+            }
+
+            if (reader.GetString() != propertyName)
+            {
+                return false;
+            }
+
+            reader.Read();
+            switch (type) {
+                case InputPrimitiveType primitype:
+                    switch (primitype.Kind)
+                    {
+                        case InputTypeKind.String:
+                            value = reader.GetString() ?? throw new JsonException();
+                            break;
+                        case InputTypeKind.Uri:
+                            var stringvalue = reader.GetString() ?? throw new JsonException();
+                            value = new Uri(stringvalue);
+                            break;
+                        case InputTypeKind.Int32:
+                            value = reader.GetInt32();
+                            break;
+                        case InputTypeKind.Int64:
+                            value = reader.GetInt64();
+                            break;
+                        case InputTypeKind.Boolean:
+                            value = reader.GetBoolean();
+                            break;
+                        default:
+                            value = reader.GetString() ?? throw new JsonException();
+                            break;
+
+                    }
+                    break;
+                case InputModelType model:
```

Comment:
I suggest we just throw in this case (except if value is `null`). We have no design for default values of the properties which type is a user-defined model.

Reply:
updated